packager/mux/mux_test.go

```diff
@@ -0,0 +1,199 @@
+			params{}},
+		{"signer",
+			" with query, empty",
+			Expand("$HOST/priv/doc?"),
+			params{}},
+		{"signer",
+			" with query, regular",
+			Expand("$HOST/priv/doc?fetch=$FETCH&sign=$SIGN"),
+			params{}},
+		{"signer",
+			" with query, escaping",
+			Expand("$HOST/priv/doc?fetch=$FETCH&sign=$SIGN%2A\\"),
+			params{}},
+		{"signer",
+			" with path, empty",
+			Expand("$HOST/priv/doc/"),
+			params{"signURL": ""}},
+		{"signer",
+			"  with path, regular",
+			Expand("$HOST/priv/doc/$FETCH"),
+			params{"signURL": Expand("$FETCH")}},
+		{"signer",
+			" with path, escaping",
+			Expand("$HOST/priv/doc/$FETCH%2A\\"),
+			params{"signURL": Expand("$FETCH%2A%5C")}},
+		{"signer",
+			" with path and query, regular",
+			Expand("$HOST/priv/doc/$FETCH?amp=1"),
+			params{"signURL": Expand("$FETCH?amp=1")}},
+		{"signer",
+			" with path and query, escaping",
+			Expand("$HOST/priv/doc/$FETCH%2A\\?amp=1%2A\\"),
+			params{"signURL": Expand("$FETCH%2A%5C?amp=1%2A\\")}},
+		{"cert",
+			" empty",
+			Expand("$HOST/amppkg/cert/"),
+			params{"certName": ""}},
+		{"cert",
+			" regular",
+			Expand("$HOST/amppkg/cert/$CERT"),
+			params{"certName": Expand("$CERT")}},
+		{"cert",
+			" escaping",
+			Expand("$HOST/amppkg/cert/$CERT%2A\\"),
+			params{"certName": Expand("$CERT*\\")}},
+		{"validityMap",
+			" regular",
+			Expand("$HOST/amppkg/validity"),
+			params{}},
+		{"healthz",
+			" regular",
+			Expand("$HOST/healthz"),
+			params{}},
+	}
+	for _, tt := range tests {
+		testName := tt.expectedHandlerName + tt.testScenario
+		t.Run(testName, func(t *testing.T) {
+			mocks := map[string](*MockedHandler){"signer": &MockedHandler{}, "healthz": &MockedHandler{}, "cert": &MockedHandler{}, "validityMap": &MockedHandler{}}
+			mux := New(mocks["cert"], mocks["signer"], mocks["validityMap"], mocks["healthz"])
+
+			// Set expectation.
+			expectedMockedHandler := mocks[tt.expectedHandlerName]
+			expectedMockedHandler.On("ServeHTTP", tt.expectedParsedParams)
+
+			// Run.
+			actualResp := pkgt.Get(t, mux, tt.url)
+
+			// Expect no errors.
+			assert.Equal(t, 200, actualResp.StatusCode, "No error expected: %#v", actualResp)
+
+			// Expect the right call to the right handler, and no calls to the rest.
+			for _, mockedHandler := range mocks {
+				mockedHandler.AssertExpectations(t)
+			}
+		})
+	}
+}
+
+func ExpectError(t *testing.T, url string, expectedErrorMessage string, expectedErrorCode int, body io.Reader) {
+	// Initialize mux with 4 identical mocked handlers, because no calls are expected to any of them.
+	mockedHandler := new(MockedHandler)
+	mux := New(mockedHandler, mockedHandler, mockedHandler, mockedHandler)
+
+	// Run and extract error.
+	actualResp := pkgt.GetBHH(t, mux, url, "", body, http.Header{})
+	actualErrorMessageBuffer, _ := ioutil.ReadAll(actualResp.Body)
+	actualErrorMessage := fmt.Sprintf("%s", actualErrorMessageBuffer)
+
+	// Expect the right error.
+	assert.Equal(t, expectedErrorCode, actualResp.StatusCode,
```

Comment:
go/go-test-comments#identify-the-function

Reply:
I've removed the custom error altogether, since testify asserts provide verbose output already:

```
-- FAIL: TestServeHTTPexpect404s (0.00s)
    --- FAIL: TestServeHTTPexpect404s/No_such_endpoint______________________ (0.00s)
    	mux_test.go:171: 
    			Error Trace:	mux_test.go:171
    			            				mux_test.go:192
    			Error:      	Not equal: 
    			            	expected: 405
    			            	actual  : 404
    			Test:       	TestServeHTTPexpect404s/No_such_endpoint______________________
    --- FAIL: TestServeHTTPexpect404s/Signer_-_unexpected_extra_char________ (0.00s)
    	mux_test.go:171: 
    			Error Trace:	mux_test.go:171
    			            				mux_test.go:192
    			Error:      	Not equal: 
    			            	expected: 405
    			            	actual  : 404
    			Test:       	TestServeHTTPexpect404s/Signer_-_unexpected_extra_char________
```